### ISSUE TITLE
IcDatePicker - Accessibility WCAG 1.4.13

### ISSUE THREAD

**Issue by MI6-55** [mi6/ic-ui-kit#2090]
# Issue
- When pressing esc while focussed on any part of the component focus moves to the calendar icon - **this is incorrect behaviour**
- When not focussed on the element (i.e you have not clicked any part of it) but hover over the pointer over the calendar icon a tooltip appears and is dismissible via esc - **this is correct behaviour**
- When focussed on any part of the element (i.e click 'DD') move the pointer to hover over the calendar icon, it is not dismissible via esc - **This is incorrect behaviour.**
- When focussed on any part of the element (i.e tab to 'DD') then tab to the calendar icon, it is now not dismissible via esc - **This is incorrect behaviour**
## WCAG Failures
This causes the component to fail [1.4.13](https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html) due to the tooltip not always being dismissible.
